code/modules/random_map/drop/droppod_doors.dm

```diff
@@ -13,24 +13,34 @@
 /obj/structure/droppod_door/Initialize(mapload, var/autoopen)
 	. = ..(mapload)
 	if(autoopen)
-		spawn(10 SECONDS)
-			deploy()
+		deploy(null, 10 SECONDS)
 
 /obj/structure/droppod_door/attack_ai(var/mob/user)
 	if(!user.Adjacent(src))
 		return
 	attack_hand(user)
 
 /obj/structure/droppod_door/attack_hand(var/mob/user)
-	if(deploying) return
-	to_chat(user, "<span class='danger'>You prime the explosive bolts. Better get clear!</span>")
-	sleep(30)
-	deploy()
+	if(deploying)
+		return ..()
+	deploy(user, 3 SECONDS)
+	return TRUE
+
+/obj/structure/droppod_door/proc/deploy(var/mob/user, var/delay)
+	set waitfor = FALSE
 
-/obj/structure/droppod_door/proc/deploy()
 	if(deployed)
 		return
 
+	if(user)
+		to_chat(user, SPAN_DANGER("You prime the explosive bolts. Better get clear!"))
```

Comment:
The bolts can be primed multiple times because `deployed` isn't set until after the delay. `deploying` should probably be set here, since it's already checked and isn't set anywhere else.